### DISCUSSION TITLE
How to run a service in a background within a script?

### DISCUSSION THREAD

**Discussion by OliverPereira** [tektoncd/pipeline#8140] [Q&A] [answered]
Hi,

We are in the process of migrating our pipelines from Gitlab to Tekton.

One of the stages (Cypress test) require the service to start up and run in the background. This works fine in Gitlab using the ampersand (&) sign but the same doesn't work in the Tekton script task.

Is this a bug or is there a alternative way to start a service in the background?

**Comment by chitrangpatel** [comment, marked as answer]
Hi @OliverPereira, 

The Steps in Tekton run in individual containers. So starting a service in one step and expecting it to be working in the next one does not work.

Tekton suggests the use of [Sidecars](https://tekton.dev/docs/pipelines/taskruns/#specifying-sidecars). Sidecars run alongside each Step.
